libraries/rush-lib/src/logic/operations/OperationExecutionManager.ts

```diff
@@ -333,25 +366,22 @@ export class OperationExecutionManager {
             colors.yellow(`"${name}" completed with warnings in ${record.stopwatch.toString()}.`)
           );
         }
-        // Legacy incremental build, if asked, prevent skip in dependents if the operation executed.
-        blockSkip ||= !this._changedProjectsOnly;
         this._hasAnyNonAllowedWarnings = this._hasAnyNonAllowedWarnings || !runner.warningsAreAllowed;
         break;
       }
     }
 
     // Apply status changes to direct dependents
     for (const item of record.consumers) {
-      if (blockCacheWrite) {
-        item.runner.isCacheWriteAllowed = false;
-      }
-
-      if (blockSkip) {
-        item.runner.isSkipAllowed = false;
+      if (status !== OperationStatus.RemoteExecuting) {
+        // Remove this operation from the dependencies, to unblock the scheduler
+        item.dependencies.delete(record);
       }
+    }
 
-      // Remove this operation from the dependencies, to unblock the scheduler
-      item.dependencies.delete(record);
+    if (record.status !== OperationStatus.RemoteExecuting) {
+      // If the operation was not remote, then we can notify queue that it is complete
+      this._executionQueue.complete(record);
     }
```

Comment:
After swapping the above nesting, this can merge with the block above it, since they have the same condition.